tracing-subscriber/src/layer.rs

```diff
@@ -0,0 +1,420 @@
+use tracing_core::{
+    metadata::Metadata,
+    span,
+    subscriber::{Interest, Subscriber},
+    Event,
+};
+
+use std::any::TypeId;
+
+/// A composable handler for `tracing` events.
+///
+/// The [`Subscriber`] trait in `tracing-core` represents the _complete_ set of
+/// functionality required to consume `tracing` instrumentation. This means that
+/// a single `Subscriber` instance is a self-contained implementation of a
+/// complete strategy for collecting traces; but it _also_ means that the
+/// `Subscriber` trait cannot easily be composed with other `Subscriber`s.
+///
+/// In particular, `Subscriber`s are responsible for generating [span IDs] and
+/// assigning them to spans. Since these IDs must uniquely identify a span
+/// within the context of the current trace, this means that there may only be
+/// a single `Subscriber` for a given thread at any point in time &mdash;
+/// otherwise, there would be no authoritative source of span IDs.
+///
+/// On the other hand, the majority of the `Subscriber` trait's functionality
+/// is composable: any number of subscribers may _observe_ events, span entry
+/// and exit, and so on, provided that there is a single authoritative source of
+/// span IDs. The `Layer` trait represents this composable subset of the
+/// `Subscriber` behavior; it can _observe_ events and spans, but does not
+/// assign IDs.
+///
+/// [`Subscriber`]: https://docs.rs/tracing-core/0.1.1/tracing_core//subscriber/trait.Subscriber.html
```

Comment:
wouldn't linking locally be better? like `../../tracing-core/...`

Reply:
ah, yeah, good call; i hadn't thought of that!

Reply:
Hmm, due to the way that docs.rs formats crate names in URLs I am not sure if it's possible to do these links locally in a way that works with both docs.rs and local doc builds. I think we may have to use absolute links.